### DISCUSSION TITLE
Papermerge 3.0.2 | Activate Italian as OCR language

### DISCUSSION THREAD

**Discussion by falbalae** [ciur/papermerge#585] [Q&A]
Hello! 
I have just finished installing Papermerge 3.0.2 and when I start OCR it asks me which language I want to use, which is, either German or English. Is there any chance to select Italian as well?

The documentation https://docs.papermerge.io/3.0/user/ocr/#default-ocr-language makes reference to a path (User Menu > Preferences > OCR > Language) which I do not find on my installation. As you may see in the printscreen below, the user menu allows me to create new users, it does not allow me configure anything else. Or if it was meant to click on my user name, I can select only "REST API" and "About"). 

![UserMenu](https://github.com/ciur/papermerge/assets/127622045/2963e93b-bac4-4c4c-b1c8-0d0a82aa6d17)

By double checking the documentation I see that at the following link https://docs.papermerge.io/3.0/settings/ocr/#ocr__default_language it is specified how to command the application to use a given language as a default. But where whould I write such command? Where do I find the configuration file? 

And all of the above it to set Italian as default language. But what if I just wanted it to have it available on the drop down list together with English and German? 

![RunOCR](https://github.com/ciur/papermerge/assets/127622045/12b0e4ac-31b0-4048-a1f1-88026dbe6d82)

Thanks.

**Comment by ciur** [comment]
You need to follow procedure described here: https://docs.papermerge.io/3.0/setup/add-ocr-langs/

Documentation here is outdated and confusing, because as per doc above it implies that "French, Italian, Spanish" are already installed, which is only partially true. 

Anyway, just create new docker image as [described here](https://docs.papermerge.io/3.0/setup/add-ocr-langs/), with italian tesseract package  (`tesseract-ocr-ita`) and you will have Italian language for OCRing.

Thank you for pointing outdated documentation references as it helps a lot to maintain documentation up to date!

**Reply by thndrbck** [reply]
I'm using a dockerfile to deploy Papermerge. In the dockerfile the syntax for setting the default OCR language environment variable is:
        PAPERMERGE__OCR__DEFAULT_LANGUAGE: eng
The delimiter is ":"
In the documentation on this page: https://docs.papermerge.io/3.2/settings/ocr/
you have the environment variable set with a different delimiter "="
PAPERMERGE__OCR__DEFAULT_LANGUAGE=spa
I don't know what context that syntax would be used. perhaps command line.
hope this helps

**Comment by ciur** [comment]
I opened [a ticket](https://github.com/ciur/papermerge/issues/586) which describes the problem. Also I fixed it.
Fix will be available as part of 3.0.3 release.

In short, there was indeed a problem that even though docs mentioned that Italian (Spanish etc) are shipped by default, in fact docker image did not contain debian packages with respective languages OCR data.

Also I've removed from the docs outdated refs.

Thank again for taking you time for opening this discussion!

**Comment by falbalae** [comment]
Dear Euegen,

Thank you very much for your feedback, really appreciated.  

I have installed papermerge on a Synology NAS by copy pasting what mariuhosting suggests therefore I need to see if I get to understand how to modify the docker command. It shouldn't be that difficult, but I have never done it, I do not know how it works.  

The install I have done is via portainer.  

Anyhow I really appreciate your work, kudos to you.  

Just another thing I have noticed: whenever I try to add a new user nothing happens. It does not appear and I don't even get an error message whatsoever.  

Thanks a lot,

Andrea  


On 28.01.2024, 13:21:08, Eugen Ciur  ***@***.***>  wrote:

>
> I opened  a ticket (https://github.com/ciur/papermerge/issues/586)  which describes the problem. Also I fixed it.
>   
Fix will be available as part of 3.0.3 release.
